js/kucoinfutures.js

```diff
@@ -0,0 +1,1287 @@
+                'expiry': this.safeValue (market, 'expireDate'),
+                'contractSize': undefined, // TODO
+                'limits': limits,
+                'info': market,
+                // Fee is in %, so divide by 100
+                'fees': this.safeValue (this.fees, 'type', {}),
+            });
+        }
+        return result;
+    }
+
+    async fetchCurrencies (params = {}) {
+        // TODO: Emulate?
+        return undefined;
+    }
+
+    async fetchTime (params = {}) {
+        const response = await this.futuresPublicGetTimestamp (params);
+        //
+        //    {
+        //        code: "200000",
+        //        data: 1637385119302,
+        //    }
+        //
+        return this.safeNumber (response, 'data');
+    }
+
+    async fetchOHLCV (symbol, timeframe = '15m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const marketId = market['id'];
+        const request = {
+            'symbol': marketId,
+            'granularity': this.timeframes[timeframe],
+        };
+        const duration = this.parseTimeframe (timeframe) * 1000;
+        let endAt = this.milliseconds ();
+        if (since !== undefined) {
+            request['from'] = since;
+            if (limit === undefined) {
+                limit = this.safeInteger (this.options, 'fetchOHLCVLimit', 200);
+            }
+            endAt = this.sum (since, limit * duration);
+        } else if (limit !== undefined) {
+            since = endAt - limit * duration;
+            request['from'] = since;
+        }
+        request['to'] = endAt;
+        const response = await this.futuresPublicGetKlineQuery (this.extend (request, params));
+        //
+        //    {
+        //        "code": "200000",
+        //        "data": [
+        //            [1636459200000, 4779.3, 4792.1, 4768.7, 4770.3, 78051],
+        //            [1636460100000, 4770.25, 4778.55, 4757.55, 4777.25, 80164],
+        //            [1636461000000, 4777.25, 4791.45, 4774.5, 4791.3, 51555]
+        //        ]
+        //    }
+        //
+        const data = this.safeValue (response, 'data', []);
+        since = this.safeString (since);
+        return this.parseOHLCVs (data, market, timeframe, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        //    [
+        //        "1545904980000",          // Start time of the candle cycle
+        //        "0.058",                  // opening price
+        //        "0.049",                  // closing price
+        //        "0.058",                  // highest price
+        //        "0.049",                  // lowest price
+        //        "0.018",                  // base volume
+        //        "0.000945",               // quote volume
+        //    ]
+        //
+        return [
+            Precise.stringDiv (this.safeString (ohlcv, 0), '1000'),
```

Comment:
The timestamp needs to be an integer, you can use `this.safeInteger` here

Reply:
stringDiv doesn't work if one of them is an integer

Reply:
i don't think we need to divide here, the timestamp looks like it is already in milliseconds